tutorials/llm/slimpajama/data_pipeline.ipynb

```diff
@@ -0,0 +1,328 @@
+    "> This customization is handy if you already have `jsonl` files processed, for example, from NeMo-Curator."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's define a local executor to run the experiment locally.\n",
+    "def docker_executor(host_data_path: str):\n",
+    "    packager = run.GitArchivePackager(subpath=\"examples/llm/slimpajama\") # This will package all code inside the folder. NOTE: only committed changes are packaged, so if you make a change, make sure to commit it.\n",
+    "    executor = run.DockerExecutor(\n",
+    "        packager=packager,\n",
+    "        ipc_mode=\"host\",\n",
+    "        shm_size=\"30g\",\n",
+    "        env_vars={\"PYTHONUNBUFFERED\": \"1\"},\n",
+    "        volumes=[f\"{host_data_path}:/data\"],\n",
+    "        container_image=\"python:3.11\",\n",
+    "        ulimits=[\"memlock:-1\", \"stack:67108864\"],\n",
+    "    )\n",
+    "    return executor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "executor = docker_executor(host_data_path=\"/home/hemild/dev/data\")\n",
+    "with run.Experiment(\"slimpajama-data-pipeline\") as exp:\n",
+    "    exp.add(download_task, name=\"download_slimpajama\", executor=executor)\n",
+    "\n",
+    "    # Use NeMo image for the remaining tasks\n",
+    "    executor.container_image = \"nvcr.io/nvidia/nemo:dev\"\n",
+    "    exp.add(extract_task, name=\"extract_slimpajama\", executor=executor)\n",
+    "\n",
+    "    # examples/llm/slimpajama is automatically mounted to /nemo_run/code\n",
+    "    exp.add(concat_task, name=\"concat_slimpajama\", executor=executor)\n",
+    "    exp.add(preprocess_task, name=\"preprocess_slimpajama\", executor=executor)\n",
+    "\n",
+    "    exp.run(sequential=True, tail_logs=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the experiment runs succesfully, you will see the bin and idx files as shown below. These can directly be used in NeMo and Megatron Data Loaders."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "concatenated_chunk1.jsonl_text_document.bin\n",
+      "concatenated_chunk1.jsonl_text_document.idx\n"
+     ]
+    }
+   ],
+   "source": [
+    "!ls {HOST_DATA_PATH}/slimpajama_megatron"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "You can also run the same experiment on a remote cluster like Slurm by replacing the docker executor with a slurm executor. A sample definition of slurm executor looks like\n",
```

Comment:
fix capitalization and punctuation (suggest "looks like the following:")

You can also run the same experiment on a remote cluster like Slurm by replacing the docker executor with a Slurm executor. A sample definition of Slurm executor looks like the following: